### DISCUSSION TITLE
Strange pixel on border #9 - Tesseract

### DISCUSSION THREAD

**Discussion by rafehimself** [cssbattle/forum#124] [Q&A] [answered]
I got this pixel wide shadow on a border of the rectangle, can someone explain to me from where does it come from? why this is happening ?
![image](https://user-images.githubusercontent.com/58576132/136532905-e0cb865f-9991-4a55-a102-2ea9fe016cd0.png)
if I remove 3th p it's gone strangely
![image](https://user-images.githubusercontent.com/58576132/136533083-c02c015c-06d8-44c8-bf7c-8c0673e24c51.png)

![image](https://user-images.githubusercontent.com/58576132/136533658-69230745-480e-44be-823e-d41fea578609.png)
main code

`<p><p><p><style>*{background:var(--,#222730);margin:0}
  p{
    height:150;
    --:#4caab3;
    margin:75 0
  }
  p+p{width:150;
height:150;
transform:rotate(45deg);
position:fixed;
top:-50;left:75;
border:53q solid#222730}
 p+p+p{--:#393E46;border-radius:50%;height:50;width:50;border:0;left:175;top:50}`

**Comment by tenderking** [comment]
i had to use outline instead of border. don't know why its happens though

**Comment by SelenIT** [comment, marked as answer]
Such a thin line often is produced by the background of the element: by default, the background fills the whole area of the element, including the area under its border, and when the element is rotated or clipped by the non-linear shape, the anti-aliasing of the edge makes some pixels of the border at the edge of the element semi-transparent, so the background "shines through" them. The typical workaround is using alternatives to `border` that are painted outside the background area (usually `outline` or `box-shadow`). But regarding that particular target, it might be useful to remember that `background` can specify both color and image (gradient) in the same time, so... maybe an extra property is not needed at all? :)
